### DISCUSSION TITLE
IIS point to wrong certificate

### DISCUSSION THREAD

**Discussion by leantalent** [GeorgeSchiro/GetCert2#18] [Q&A] [answered]
I have run getcert with the old config to a new domain..... apparently there were setting for all the other domains I have run in the script.bat
1. could not do a new certificate but I got an error in the stage 5
2022-09-19 11:29:56:045 AM  Stage 5 - generate certificate request and
submit ...
2022-09-19 11:29:56:077 AM
2022-09-19 11:29:56:092 AM  Import-Module
"C:\Users\catalin\Desktop\GetCert2020\ACME-PS"
[Net.ServicePointManager]::SecurityProtocol =
[Net.SecurityProtocolType]::Tls12
$order = Find-ACMEOrder "C:\Users\c\Desktop\GetCert2020\AcmeState"
-Identifiers ("mta-sts.mindmaster.ro")
$certKey = New-ACMECertificateKey -Path
"C:\Users\catalin\Desktop\GetCert2020\AcmeState\cert.key.xml"
Complete-ACMEOrder "C:\Users\c\Desktop\GetCert2020\AcmeState"
-Order $order
2022-09-19 11:29:58:155 AM  Server returned Problem (Status:
403).A'nOrder's status ("invalid") is not acceptable for finalization
2022-09-19 11:29:58:155 AM  At
C:\Users\c\Desktop\GetCert2020\ACME-PS\1.1.5\ACME-PS.psm1:1401 char:13
2022-09-19 11:29:58:155 AM  +             throw
[AcmeHttpException]::new($response.ErrorMessage, $r ...
2022-09-19 11:29:58:155 AM  +
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-09-19 11:29:58:155 AM      + CategoryInfo          :
OperationStopped: (:) [], AcmeHttpException
2022-09-19 11:29:58:155 AM      + FullyQualifiedErrorId : Server
returned Problem (Status: 403).A'nOrder's status ("invalid") is not
acceptable fo
2022-09-19 11:29:58:155 AM     r finalization
2022-09-19 11:29:58:155 AM
2022-09-19 11:29:58:170 AM  GetCertServiceFault: The sub-process
experienced a critical failure.
2022-09-19 11:29:58:186 AM
2022-09-19 11:29:58:217 AM  At least one stage failed (or the process
was stopped). Check log for errors.
2022-09-19 11:29:58:233 AM

2. all the domains are now pointing to just one certificate even that in IIS I see that there is the correct certificate selected. 
Attackers might be trying to steal your information from mydomain.com (for example, passwords, messages, or credit cards).
NET::ERR_CERT_COMMON_NAME_INVALID

Let me know first if point 2 is any solution....

**Comment by GeorgeSchiro** [comment, marked as answer]
Whenever you see something like this in stage 5:

    Server returned Problem (Status: 403). A'nOrder's status ("invalid") is not acceptable for finalization

that implies an issue with this:

    Stage 4 - update challenge from certificate provider ...

So the problem is really before stage 5.

Also, whenever you have a problem with the domain name challenge, look for the HTTP challenge URL in stage 3.

It's not incuded in what you posted. It should look like this:

    [your domain here]/.well-known/acme-challenge/JjHGZmKBXnpQMWP3XhfnbsedDlny13fk4l8t26H6j5U

    (BTW, the above URL should work "as is" in your browser with or without an "http://" prefix.) 

Paste the challenge URL into a browser to see what comes up. If you get any sort of error, that would explain why Let's Encrypt is declaring the challenge "invalid".

**Reply by leantalent** [reply]
OK, point 1 it is fixed.
What I do that all my domain2.com domain3.com all point to the certificate for domain1.com then one I have also mts-sts.domain1.com.?

**Comment by leantalent** [comment]
actually now I see that the domains point to the newly created mts-sts.domain1.com :(

**Comment by leantalent** [comment]
what I see in the config is this:
Bind certificate in IIS ...
Applying new certificate for &quot;mta-sts.domain1.com&quot; ...
New certificate (&quot;4826E076AF0917CCA9E26D62E242ADC3AE3E06B7&quot;) bound to address 00.00.00.00 and port 443 in IIS for &quot;mta-sts.domain1.com&quot;.
Success

which means that the IP get the same certificate.
How I change back so each domain will point to its certificate?

**Comment by GeorgeSchiro** [comment]
I'm not sure I understand the question Catalin.

Are you asking how to manually bind your certificate to various sites? Or do you want to prevent part or all of the automated binding that GetCert does?

**Comment by leantalent** [comment]
for the moment all sites get same certificate, with or without SNI .... and I need to fix this manually

**Comment by leantalent** [comment]
I do not know what it was the problem, but I have set all websites to have a single IP and SNI box checked, and all works fine now....

**Comment by GeorgeSchiro** [comment]
If you are unsure of your automated process - and assuming you run it every day, I would recommend generating a new certificate during off hours (like on a weekend) and then set -RenewalDaysBeforeExpiration=83 .

That way you can test weekly until the automated process is reliably working. You can also monitor the certificate replacement process and fix any problem when it will be less trouble for your business.
